README.md

```diff
@@ -270,8 +277,44 @@ It is perfectly fine to chain multiple `views`, `props` calls etc in arbitrary o
 
 It is also possible to define lifecycle hooks in the _actions_ object, these are actions with a predefined name that are run at a specific moment. See [Lifecycle hooks](#lifecycle-hooks-for-typesmodel).
 
-_Tip: `(self) => ({ action1() { }, action2() { }})` is ES6 syntax for `function (self) { return { action1: function() { }, action2: function() { } }}`, in other words; it's short way of directly returning an object literal.
-For that reason a comma between each member of a model is mandatory, unlike classes which are syntactically a totally different concept._
+TypeScript:
+
+TypeScript is supported, but you sometimes need to take into account where your typings are available.
+
+The code below will not compile: TypeScript will complain that `self.upperProp` is not a property. Computed properties are only available after `.views` is evaluated.
+
+```typescript
+const Example = types.model('Example', { prop: types.string }).views(self => ({
```

Comment:
Note that another alternative worth mentioning is, as long as there are no circular deps:

```javascript
const Example = types
  .model('Example', { prop: types.string })
  .views(self => ({
    get upperProp(): string {
      return self.prop.toUpperCase();
    }
  }))
  .views(self => ({
    get twiceUpperProp(): string {
      return self.upperProp + self.upperProp;
    }
  }));
```